imod_coupler/logging/exchange_collector.py

```diff
@@ -0,0 +1,107 @@
+import abc
+import os
+from pathlib import Path
+from typing import Any, List, Optional
+
+import netCDF4 as nc
+import numpy as np
+import tomli
+from numpy.typing import NDArray
+from typing_extensions import Self
+
+
+class AbstractExchange(abc.ABC):
+    def __init__(self, name: str):
+        pass
+
+    def write_exchange(self, exchange: NDArray[Any], time: float) -> None:
+        pass
+
+    def finalize(self) -> None:
+        pass
+
+
+class NetcdfExchangeLogger(AbstractExchange):
+    output_file: Path
+    name: str
+
+    def __init__(self, name: str, output_dir: Path, properties: dict[str, Any]):
+        if not (os.path.isdir(output_dir)):
+            os.mkdir(output_dir)
```

Comment:
You can use `pathlib` instead of `os`.
https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_dir
https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir